src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -302,6 +307,14 @@ bool ov::pass::ConvertPrecision::run_on_model(const std::shared_ptr<ngraph::Func
         {opset10::Unique::get_type_info_static(), fuse_type_to_unique_v10},
         {opset8::RandomUniform::get_type_info_static(), fuse_type_to_random_uniform_v8}};
 
+    if (m_keep_precision_sensitive_in_fp32) {
+        pass::Manager manager(get_pass_config());
+        // Mark subgraphs with disable_fp16_compression to keep them in FP32
+        manager.register_pass<pass::MarkSugraphsToKeepInMixedPrecision>();
+        manager.register_pass<pass::AlignMixedFP32FP16Types>();
```

Comment:
As the passes above are required for fp32->fp16 conversion only, I think we may add a check that such pair exists in precisions list and don't run redundant transforms if not really needed.

Reply:
agreed. will do that

Reply:
done